conans/model/conan_file.py

```diff
@@ -285,3 +290,65 @@ def __repr__(self):
             return "%s/%s@PROJECT" % (self.name, self.version)
         else:
             return "PROJECT"
+
+
+class ConanFileEditable(object):
```

Comment:
I feel this fake inheritance is a bit difficult to understand. As the whole point of it is just doing some extra post-processing after ``package_info()`` is called, I think I would implement directly that: in ``ConanInstaller._call_package_info()`` for those editable packages, one extra thing is needed. The package layout file will be there (package_folder available), so it can be loaded and used.